lighthouse-core/gather/gatherers/seo/tap-targets.js

```diff
@@ -287,17 +346,66 @@ function gatherTapTargets() {
       return;
     }
 
-    targets.push({
-      clientRects: visibleClientRects,
-      snippet: truncate(tapTargetElement.outerHTML, 300),
-      // @ts-ignore - getNodePath put into scope via stringification
-      path: getNodePath(tapTargetElement),
-      // @ts-ignore - getNodeSelector put into scope via stringification
-      selector: getNodeSelector(tapTargetElement),
-      href: /** @type {HTMLAnchorElement} */(tapTargetElement)['href'] || '',
+    const largestRect = getLargestRect(visibleClientRects);
+    const largestRectCenterPoint = getRectCenterPoint(largestRect);
+    // round so we can can assume whole numbers during in-viewport check
+    largestRectCenterPoint.x = Math.round(largestRectCenterPoint.x);
+    largestRectCenterPoint.y = Math.round(largestRectCenterPoint.y);
+
+    if (largestRectCenterPoint.x >= window.innerWidth) {
+      // we don't scroll sideways, so the center of this tap target is always hidden
+      return;
+    }
+
+    enhancedTapTargets.push({
+      tapTargetElement,
+      largestRectCenterPoint,
+      visibleClientRects,
     });
   });
 
+
+  enhancedTapTargets.sort(
+    (a, b) => {
+      return a.largestRectCenterPoint.y - b.largestRectCenterPoint.y;
+    }
+  );
+
+  // Disable point events so that tap targets below them don't get
+  // detected as non-tappable (they are tappable, just not while the viewport
+  // is at the current scroll position)
+  const reenableFixedAndStickyElementPointerEvents = disableFixedAndStickyElementPointerEvents();
+
+  let item;
+  while (item = enhancedTapTargets.shift()) {
+    const {tapTargetElement, largestRectCenterPoint, visibleClientRects} = item;
+
+    // todo: do soething to prevent infinite loop here
+    while (!pointIsInViewport(largestRectCenterPoint)) {
+      const finalScrollPos = document.documentElement.scrollHeight;
+      if (document.documentElement.scrollTop >= finalScrollPos) {
+        throw Error('scrolled all the way but not found');
+      }
+      document.documentElement.scrollTop += window.innerHeight;
+    }
+
+    const isTop = elementCenterIsAtZAxisTop(tapTargetElement, largestRectCenterPoint);
```

Comment:
so this is really beyond just working with absolute elements this is about making sure we're only looking at what's on top for *all* elements, right?

Reply:
It is right now, but just because I forgot to just do the check for position absolute elements. I think we should filter it.

Reply:
Just realized that we need to handle full-page app install/newsletter/GDPR overlays...

We could check if all elements are top layer, not just position absolute ones. Then if you have a full-page overlay you get an easy 100% score.

Or we could only check position absolute elements, but then we can get false failures with the content behind them.

I'm leaning towards the first option... if your site has a full-page overlay it kinda makes sense that the tap targets audit isn't very useful?

Reply:
+1 to the first approach. We should keep the signal as high as possible in this audit, and I think false positives from absolute elements would hurt that if we went with option 2.

If they have a overlay, it makes sense that we can only test what we see :)

Reply:
Sorry, what's the final state of this now then?

Reply:
- we go with option 1, so if a lot of content is covered by a position absolute element then we ignore those tap targets
- we ignore position `fixed`/`sticky` elements when calling `getElementFromPoint`, so tap targets underneath them are detected – we were doing that before but I got confused because reddit had a fixed overlay with `pointer-events: all`. My CSS rule is now `!important`, so that's fixed